go/vt/vtorc/inst/instance_dao.go

```diff
@@ -429,6 +437,8 @@ Cleanup:
 				instance.GtidErrant, err = vitessmysql.Subtract(redactedExecutedGtidSet.String(), redactedPrimaryExecutedGtidSet.String())
 			}
 		}
+		// update the errant gtid map
+		errantGtidMap[topoproto.TabletAliasString(tablet.Alias)] = instance.GtidErrant
```

Comment:
Could we be writing to and reading (the metric) from the map  concurrently? In that case we will need to synchronize the accesses.

Also could there be multiple concurrent writes to this map? I see `handleDiscoveryRequests()` is discovering instances in parallel.

Reply:
This map is only used for a metric, so it should be okay to read and write concurrently to it, right? I can add locking around it, but I don't see the need for it.

Reply:
I have never seen this in practice, but as per 1.6+ https://go.dev/doc/go1.6 says, :
> Runtime
The runtime has added lightweight, best-effort detection of concurrent misuse of maps. As always, if one goroutine is writing to a map, no other goroutine should be reading or writing the map concurrently. If the runtime detects this condition, it prints a diagnosis and crashes the program. The best way to find out more about the problem is to run the program under the [race detector](https://blog.golang.org/race-detector), which will more reliably identify the race and give more detail.

Another reference: https://stackoverflow.com/questions/36167200/how-safe-are-golang-maps-for-concurrent-read-write-operations which shows a runtime panic can occur.

I think if you had a unit test around this the race detector would have failed it.

Reply:
Agreed with @rohit-nayak-ps, this requires synchronization.

Reply:
Added.